src/pixelator/analysis/polarization/__init__.py

```diff
@@ -32,51 +30,59 @@
 logger = logging.getLogger(__name__)
 
 
-def morans_autocorr(w: np.ndarray, y: np.ndarray) -> Any:
-    """
-    A helper function that computes the Moran's I autocorrelation statistics
-    for the given spatial weights (w) and target variable (y). The target
-    variable can be the counts of a specific antibody so the Moran's I value
-    could indicate if the antibody has a localized spatial pattern. The function
-    returns an object with the Moran's statistics (I, p_rand and z_rand).
+def morans_autocorr(
+    w: np.ndarray, y: np.ndarray, permutations: int
+) -> esda.moran.Moran:
+    """Calculate Moran's I statistics.
+
+    Computes the Moran's I autocorrelation statistics for the given spatial
+    weights (`w`) and target variable (`y`). The target variable can be the counts
+    of a specific antibody so the Moran's I value could indicate if the
+    antibody has a localized spatial pattern. The function returns an object
+    with the Moran's statistics: I, p_rand and z_rand (as well as p_sim and z_sim
+    if `permutations` > 0).
     :param w: the weights matrix (connectivity matrix)
     :param y: the counts vector
+    :param permutations: the number of permutations for simulated Z-score (z_sim)
+                         estimation (if permutations > 0)
     :returns: the Moran's statistics
+    :rtype: esda.moran.Moran
     """
     # Default Moran's transformation is row-standardized "r".
     # https://github.com/pysal/esda/blob/main/esda/moran.py
-    return Moran(y, w, permutations=None)
+    return Moran(y, w, permutations=permutations)
 
 
 def polarization_scores_component(
     graph: Graph,
     component_id: str,
-    normalization: Literal["raw", "clr", "denoise"] = "clr",
-    antibody_control: Optional[List[str]] = None,
-    binarization: bool = False,
+    normalization: Literal["raw", "clr"] = "clr",
```

Comment:
Literal could be rewritten as `PolarizationNormalizationTypes`, but remember to import it in the header of the file.